### PR TITLE
impl(GCS+gRPC): build tweaks for gRPC OTel plugin

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #13998

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14251)

### CODE REVIEW

File: google/cloud/storage/BUILD.bazel [hunk 1]
```diff
@@ -46,7 +46,7 @@ cc_library(
     srcs = google_cloud_cpp_storage_grpc_srcs,
     hdrs = google_cloud_cpp_storage_grpc_hdrs,
     defines = ["GOOGLE_CLOUD_CPP_STORAGE_HAVE_GRPC"],
-    local_defines = select({
+    local_defines = ["GOOGLE_CLOUD_CPP_GRPC_HAS_OTEL_PLUGIN"] + select({
```
Comment (dbolduc) [inline google/cloud/storage/BUILD.bazel:49]: Just curious, now that we are making this thing optional for CMake, will you do the same for Bazel?

If you go for the opt-out flag, the `deps` on this target will need to change. (That might be complicated enough to warrant separate CMake and Bazel PRs)

**Review by dbolduc — COMMENTED** [review]
I think this would also need to be conditional:

https://github.com/googleapis/google-cloud-cpp/blob/1fb897c149284f236a9b40469200d769f8f13a29/cmake/GoogleCloudCppFeatures.cmake#L335

(ugly, but) something like:

```
find_package(opentelemetry-cpp CONFIG)
if (opentelemetry-cpp_FOUND)
   list(INSERT GOOGLE_CLOUD_CPP_ENABLE 0 opentelemetry) 
endif ()
```

**Comment by coryan** [comment]
Back to the beginning.

**Review by dbolduc — APPROVED** [review]
Approved under duress. I think we will need to revisit these decisions. Opened #14256

**Pull request merged**